designer/server/src/common/helpers/auth/get-user-session.js

```diff
@@ -32,28 +31,37 @@ export async function getUserSession(request, session) {
 
 /**
  * @param {AuthCredentials | null} [credentials]
+ * @returns {credentials is AuthWithTokens}
```

Comment:
Should `hasAuthenticated()` return true, we narrow `AuthCredentials` to `AuthTokens`

This still leaves `user` as optional but prevents unnecessary token checks